cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/AbstractMapReduceContextBuilder.java

```diff
@@ -106,7 +107,7 @@ public BasicMapReduceContext build(MapReduceMetrics.TaskType type,
     BasicMapReduceContext context =
       new BasicMapReduceContext(program, type, RunIds.fromString(runId), taskId,
                                 runtimeArguments, datasets, spec, logicalStartTime,
-                                workflowBatch, discoveryServiceClient, metricsCollectionService,
+                                workflowBatch, adapterName, discoveryServiceClient, metricsCollectionService,
```

Comment:
when its used outside of adapters, the ``adapterName`` would be null in ``BasicMapreduceContext``. Similar pattern as ``workflowBatch``, but perhaps we need some refactoring here for ``BasicMapreduceContext`` to only receive ``Arguments`` and do the extraction that it needs within.